pkg/util/containers/filter.go

```diff
@@ -367,21 +372,25 @@ func (cf Filter) IsExcluded(containerName, containerImage, podNamespace string)
 	return false
 }
 
-// IsExcludedByAnnotation identifies whether a container should be excluded
+// isExcludedByAnnotation identifies whether a container should be excluded
 // based on the contents of the supplied annotations.
-func IsExcludedByAnnotation(ft FilterType, annotations map[string]string, containerName string) bool {
-	switch ft {
+func (cf Filter) isExcludedByAnnotation(annotations map[string]string, containerName string) bool {
+	if annotations == nil {
+		return false
+	}
+	switch cf.FilterType {
 	case GlobalFilter:
-		return isExcludedByAnnotation(annotations, containerName, "")
 	case MetricsFilter:
-		return isExcludedByAnnotation(annotations, containerName, "metrics_")
+		return isExcludedByAnnotationInner(annotations, containerName, "metrics_")
 	case LogsFilter:
-		return isExcludedByAnnotation(annotations, containerName, "logs_")
+		return isExcludedByAnnotationInner(annotations, containerName, "logs_")
+	default:
+		log.Warnf("unrecognized filter type: %s", cf.FilterType)
 	}
-	return false
+	return isExcludedByAnnotationInner(annotations, containerName, "")
```

Comment:
I noticed that we should always check for `GlobalFilter`. Otherwise, the `SharedMetricFilter` (retrieved with `GetSharedMetricFilter()` will not filter out corecheck metrics.